metrics/core/src/main/java/org/hyperledger/besu/metrics/opentelemetry/DebugMetricReader.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Besu Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.metrics.opentelemetry;
+
+import java.util.Collection;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.CollectionRegistration;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
+import io.opentelemetry.sdk.metrics.internal.export.MetricProducer;
+import org.jetbrains.annotations.NotNull;
+
+class DebugMetricReader implements MetricReader {
+  private CollectionRegistration registration;
+
+  public DebugMetricReader() {}
+
+  public Collection<MetricData> getAllMetrics() {
+    return MetricProducer.asMetricProducer(this.registration).collectAllMetrics();
+  }
+
+  @Override
+  public void register(final @NotNull CollectionRegistration registration) {
```

Comment:
nit: Are we using jetbrains annotations in Besu or is it auto-generated?

Reply:
It's probably a remainder of code I copied over. I removed them.